packages/radio-player/src/radio-player.ts

```diff
@@ -0,0 +1,738 @@
+    // This is a case where the functionality is different enough to have two instances
+    // of it instead of one and just show and hide them based on the media query.
+    return html`
+      <div class="search-section">
+        <expandable-search-bar
+          searchTerm=${this.searchTerm}
+          .quickSearches=${this.quickSearches}
+          @inputchange=${this.updateSearchTerm}
+          @enterKeyPressed=${this.searchEnterKeyPressed}
+          @searchCleared=${this.searchCleared}
+        >
+        </expandable-search-bar>
+        <div class="search-results-info">
+          ${this.searchResultsSwitcherTemplate} ${this.noSearchResultsTemplate}
+        </div>
+      </div>
+    `;
+  }
+
+  private get searchResultsSwitcherTemplate(): TemplateResult {
+    return html`
+      <search-results-switcher
+        class="${this.shouldShowSearchResultSwitcher ? '' : 'hidden'}"
+        @searchResultIndexChanged=${this.searchResultIndexChanged}
+      >
+      </search-results-switcher>
+    `;
+  }
+
+  private get noSearchResultsTemplate(): TemplateResult {
+    return html`
+      <div
+        class="no-search-results-message ${this.shouldShowNoSearchResultMessage ? '' : 'hidden'}"
+      >
+        No search results.
+      </div>
+    `;
+  }
+
+  private get quickSearches(): QuickSearchEntry[] {
+    if (!this.config) {
+      return [];
+    }
+
+    const entries: QuickSearchEntry[] = this.config.quickSearches.map(
+      entry => new QuickSearchEntry(entry),
+    );
+
+    return entries;
+  }
+
+  private updateSearchTerm(e: CustomEvent): void {
+    this.searchTerm = e.detail.value;
+  }
+
+  private searchCleared(): void {
+    this.searchTerm = '';
+    this.emitSearchClearedEvent();
+    /* istanbul ignore else */
+    if (this.transcriptView) {
+      this.transcriptView.selectedSearchResultIndex = 0;
+    }
+    /* istanbul ignore else */
+    if (this.searchResultsSwitcher) {
+      this.searchResultsSwitcher.currentResultIndex = 0;
+    }
+  }
+
+  private emitSearchClearedEvent(): void {
+    const event = new Event('searchCleared');
+    this.dispatchEvent(event);
+  }
+
+  private searchResultIndexChanged(e: CustomEvent): void {
+    if (!this.transcriptView) {
+      return;
+    }
+    this.transcriptView.selectedSearchResultIndex = e.detail.searchResultIndex;
+    this.transcriptView.scrollToSelectedSearchResult();
+  }
+
+  private searchEnterKeyPressed(e: CustomEvent): void {
+    const event = new CustomEvent('searchRequested', {
+      detail: { searchTerm: e.detail.value },
+    });
+    this.dispatchEvent(event);
+  }
+
+  private get transcriptEntries(): TranscriptEntryConfig[] {
+    return this.transcriptConfig ? this.transcriptConfig.entries : [];
+  }
+
+  private get transcriptView(): TranscriptView | null {
+    /* istanbul ignore next */
+    return this.shadowRoot
+      ? (this.shadowRoot.querySelector('transcript-view') as TranscriptView)
```

Comment:
I believe that `this.shadowRoot.querySelector` will return `null` if not found, so this (and others) can be shortened to `(this.shadowRoot.querySelector('transcript-view') as TranscriptView) || null`

Reply:
This is a bit of a typescript oddity that was solved in typescript 3.7 (which was just released a few days ago).

In this case (and you'll see a lot of this going on in the various components) the `shadowRoot` is nullable so you have to first check for the `shadowRoot` before you can call `querySelector` on it.

Typescript 3.7 introduced this awesome feature called optional chaining that let's you do this.. the `?` after `shadowRoot` will just let it return `null` if `shadowRoot` is `null`:

```
return this.shadowRoot?.querySelector('transcript-view') as TranscriptView
```